modules/ggl-cli/bin/ggl-cli.c

```diff
@@ -21,18 +21,30 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+#define MAX_COMPONENTS 10
+
+typedef struct {
+    char *name;
+    char *version;
+} Component;
+
 char *command = NULL;
 char *recipe_dir = NULL;
 char *artifacts_dir = NULL;
-char *component_name = NULL;
-char *component_version = NULL;
+Component components[MAX_COMPONENTS];
+int component_count = 0;
 
 static char doc[] = "ggl-cli -- Greengrass CLI for Nucleus Lite";
 
 static struct argp_option opts[] = {
     { "recipe-dir", 'r', "path", 0, "Recipe directory to merge", 0 },
     { "artifacts-dir", 'a', "path", 0, "Artifacts directory to merge", 0 },
-    { "add-component", 'c', "name=version", 0, "Component to add", 0 },
+    { "add-component",
+      'c',
+      "name=version",
+      0,
+      "Component to add (can be used multiple times)",
```

Comment:
I think argp already has something for this? Instead of adding can be used multiple times here, should update such that the docstring uses the proper `...`. I don't recall exactly how that is done.